samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -456,8 +456,13 @@ object SamzaContainer extends Logging {
     samzaContainerMetrics.containerThreadPoolSize.set(threadPoolSize)
 
     val taskThreadPool = if (threadPoolSize > 0) {
-      Executors.newFixedThreadPool(threadPoolSize,
-        new ThreadFactoryBuilder().setNameFormat("Samza Container Thread-%d").build())
+      val taskExecutorFactoryClassName = jobConfig.getTaskExecutorFactory
+      val taskExecutorFactory = ReflectionUtil.getObj(taskExecutorFactoryClassName, classOf[TaskExecutorFactory])
+      if (taskExecutorFactory != null) {
```

Comment:
This check should not be needed ReflectionUtil will throw if the class is not found/constructed